acvm/src/lib.rs

```diff
@@ -243,11 +253,22 @@ pub fn default_is_opcode_supported(
     // attempt to transform into supported gates. If these are also not available
     // then a compiler error will be emitted.
     fn plonk_is_supported(opcode: &Opcode) -> bool {
-        !matches!(
-            opcode,
-            Opcode::BlackBoxFuncCall(BlackBoxFuncCall { name: BlackBoxFunc::AES, .. })
-                | Opcode::Block(_)
-        )
+        !matches!(opcode, Opcode::BlackBoxFuncCall(BlackBoxFuncCall::AES { .. }) | Opcode::Block(_))
+    }
+
+    match language {
+        Language::R1CS => r1cs_is_supported,
+        Language::PLONKCSat { .. } => plonk_is_supported,
+    }
+}
+
+pub fn default_is_bb_supported(language: Language) -> impl Fn(BlackBoxFunc) -> bool {
```

Comment:
I'd like to avoid adding this function (see deprecation of `default_is_opcode_supported`). At the very least we should immediately deprecate it.

Reply:
https://github.com/noir-lang/acvm/pull/273/files#r1190651498

Reply:
Yeah, I don't know how we are going to check what the backend supports during compilation (at that point there is no backend associated yet)... But it's true that we can avoid adding this function by creating a dummy constructor for BB call variants. I'm going to drop this change in favor of that because it should help with the future vision of "specifying supported opcodes" instead of specifically talking about bb calls!